src/components/SettingsInputText/SettingsInputText.vue

```diff
@@ -0,0 +1,162 @@
+<!--
+  - @copyright Copyright (c) 2019 Julius Härtl <jus@bitgrid.net>
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  - @author Greta Doci <gretadoci@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<form @submit.prevent="">
+		<div class="input-wrapper">
+			<label :for="id">{{ label }}</label>
+			<input :id="id" type="text" :value="inputVal"
+				:disabled="disabled" @input="$emit('input', $event.target.value)">
+			<input type="submit" class="icon-confirm" value=""
+				@click="$emit('update', inputVal)">
+		</div>
+		<p v-if="hint !== ''" class="hint">
+			{{ hint }}
```

Comment:
Btw, maybe it would be cleaner to have the hint as the slot?
```vue
<SettingsInputText
	label="Enter password"
	v-model="password">
		Please enter a password to execute those administrative tasks
</SettingsInputText>
```